app/services/content/JavaDataCollector.java

```diff
@@ -54,15 +50,21 @@ public Source getSource() {
         String trend = trends.poll();
         List<Post> posts;
 
-        // if we have queried this trend before only get newer posts
-        if (sinceIds.containsKey(trend)) {
-            posts = source.getMaxTrendingPostsSince(trend, sinceIds.get(trend));
-        } else {
-            posts = source.getMaxTrendingPosts(trend);
-        }
+		if(source instanceof TwitterSource) {
```

Comment:
This should define general behaviour for all sources and not check the type of source. If one of the sources (in this case reddit) behaves differently then it's methods should take care of that, not the caller.

For example, the reddit source implementation of getMaxTrendingPostsSince could just directly call it's getMaxTrendingPosts. Then this class doesn't need to know anything about which source it has to work.